datasets/li/posting_sanctions/crawler.py

```diff
@@ -77,88 +75,98 @@ def parse_target(
+        address = parse_address(context, row.pop("Adresse"))
+        name = row.pop("Betrieb")
+        start = h.parse_date(row.pop("In Rechtskraft"), ["%d.%m.%Y"])
+        end = h.parse_date(row.pop("Ende der Sperre"), ["%d.%m.%Y"])
         company = parse_target(context, name, address, start)
         if company is None:
             continue
         company.add("topics", "debarment")
+        violation = row.pop("Verstoss")
         sanction = h.make_sanction(context, company)
         sanction.id = context.make_id(
-            "Sanction", "Debarment", company.id, law, start, end
+            "Sanction", "Debarment", company.id, violation, start, end
         )
-        h.apply_date(sanction, "date", start)
-        h.apply_date(sanction, "endDate", end)
+        sanction.add("startDate", start)
+        sanction.add("endDate", end)
         sanction.add("description", "Debarment")
         sanction.add("program", "EntsG Sanctions")
         reason = (
             "Repeated or severe infraction against "
-            f"Liechtenstein Posted Workers Act, {law}"
+            f"Liechtenstein Posted Workers Act, {violation}"
         )
         sanction.add("reason", reason)
         context.emit(sanction)
         context.emit(company, target=True)
 
 
-def parse_infractions(context: Context, doc) -> None:
-    table = doc.xpath(
-        "//h2[text()='Übertretungen (Art. 9 Entsendegesetz)']/following::table[1]"
-    )[0]
-    for row in table.xpath("tbody/tr")[1:]:
-        [date, name, addr, law] = row.xpath("descendant::*/text()")
-        address = parse_address(context, addr)
+def crawl_infractions(context: Context) -> None:
+    path = context.fetch_resource("uebertretungen.pdf", INFRACTION_URL)
+    for row in extract_rows(path):
+        if len(row) != 4:
+            context.log.warn(f"Cannot split row: {row}")
+            continue
+        address = parse_address(context, row.pop("Adresse"))
+        effective = row.pop("In Rechtskraft")
+        name = row.pop("Betrieb/ verantwortliche natürliche Person")
+        date = h.parse_date(effective, ["%d.%m.%Y"])
         company = parse_target(context, name, address, date)
         if company is None:
             continue
+        violation = row.pop("Verstoss")
         sanction = h.make_sanction(context, company)
-        sanction.id = context.make_id("Sanction", "Penalty", company.id, law, date)
-        h.apply_date(sanction, "date", date)
+        sanction.id = context.make_id(
+            "Sanction", "Penalty", company.id, violation, date
+        )
+        sanction.add("date", date)
         sanction.add("description", "Administrative Penalty")
         sanction.add("program", "EntsG Sanctions")
         sanction.add(
-            "reason", f"Infraction against Liechtenstein Posted Workers Act, {law}"
+            "reason",
+            f"Infraction against Liechtenstein Posted Workers Act, {violation}",
         )
         company.add("topics", "debarment")
```

Comment:
should we add reg.warn and announce a migration to that topic for this file?

Reply:
it looks a bit like art 9 is violations but not yet sanctions, but my german is a tad rustier than yours